### DISCUSSION TITLE
Img Src is not found when using dynamic src

### DISCUSSION THREAD

**Discussion by vionijannet** [nuxt/nuxt#19360] [Questions]
Using Nuxt 3.

I'm trying to load an image.
Since there is a condition which image to show, I make v-bind for the src.
```
<template>
<img :src="require(`~/assets/image/${file}`) /  >"
</template>

<script setup lang="ts">
const file="test.svg";
</script>
```
This doesn't work like any Vue project using require.

I've tried directly put filename in the src, like
`<img :src="require('~/assets/image/test.svg') />"`
but it didn't work. 

I've also tried using relative param like "../assets/image/test.svg" 
`<img :src="require('../assets/image/test.svg')" />`
but it also didn't work.

Here is the reproduction, and I don't know what seems to be missing. Or is it a bug in nuxt 3? Thanks
https://stackblitz.com/edit/nuxt-starter-6rk5en?file=app.vue

**Comment by Baroshem** [comment]
Hey,

Do you actually need to use require? In Nuxt usually, you can put the image in a public folder and reference it easily like following

```
<img :src="condition ? path1 : path2" />

```

https://github.com/Baroshem/nuxt-shopify-tailwind/blob/main/components/TheHeader.vue

**Reply by vionijannet** [reply]
So for dynamic image, the only way that can do is put in the public folder instead of asset?

**Reply by Baroshem** [reply]
I would recommend to go with public. From my experience, I usually put the stylesheets in assets and images in public :)

**Comment by TheAlexLichter** [comment]
Please follow along https://github.com/nuxt/nuxt/issues/14766 for solutions and a discussion around the Nuxt 3 / Vite behavior.

**Reply by vionijannet** [reply]
Okay, thanks for the information
